enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,345 @@
+is responsible for managing the cluster resources, users, and projects.
+
+This document provides details about how cluster administrator 
+can hide user perspectives. 
+
+![Perspectives switch dropdown](perspectives-switch.png)
+
+## Motivation
+
+Cluster administrator wants to hide the "Administrator" perspective 
+from non-privileged and to hide the "Developer" 
+perspective for all users.
+
+Currently all users have access to all perspectives and this might confuse some developers.
+
+New perspectives could be added by the customer so they want to hide the default perspectives. 
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide "Administrator" perspective 
+from all non-privileged users so that non-privileged users
+cannot access "Administrator" perspective for managing the cluster resources, 
+users, and projects. Non-privileged users are users that cannot get
+all `namespaces` (based on RBAC check)
+
+#### Story 2
+
+As a cluster admin, I want to hide "Developer" perspective
+for all users because the customers can add own perspectives and may be not interested in developer perspective.
+
+#### Story 3
+
+As a cluster admin, I want to hide any perspective contributed by a plugin.
+A plugin can add multiple features to the console, the users are may be interested 
+in other features of the plugin but not the perspective.
+
+It might be also interesting for a customer to show this new perspective
+only to users with specific permissions.
+
+Additional perspectives can be dynamically added with the 
+[Perspective extension](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/perspectives.ts).
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+- Validating the console resource in the operator and inform the cluster admins with an error because the operator
+doesn't know the available perspectives provided by the console plugins.
+- Allow cluster admins to customize the navigation menu along with perspectives for users or groups.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective id and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── visibility
+|               ├── state
+|               ├── accessReview
+|                   ├── required
+|                       ├── group
+|                       ├── resource
+|                       ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
```

Comment:
How would you expect this to be extended? Do we need to make sure the API is extendable? What's the use case?

Reply:
Sorry, this was added for missingAccessReview check. I removed that line now. That is not required.